src/Orchard.Cms.Web/Modules/Orchard.Setup/Services/SetupService.cs

```diff
@@ -135,14 +135,17 @@ ILogger<SetupService> logger
                     {
                         await store.InitializeAsync();
                     }
-                    catch
+                    catch(Exception ex)
                     {
                         // Tables already exist or database was not found
 
                         // The issue is that the user creation needs the tables to be present,
                         // if the user information is not valid, the next POST will try to recreate the
-                        // tables. The tables should be rollbacked if one of the steps is invalid,
+                        // tables. The tables should be rolled back if one of the steps is invalid,
                         // unless the recipe is executing?
+
+                        context.Errors.Add("DatabaseProvider", $"An error occurred while initializing the datastore: {ex.Message}");
+                        return null;
```

Comment:
Would he rather return an empty string instead? Little point in generating an execution ID at this point when the DB connection fails, right?